packages/react-devtools-shared/src/hook.js

```diff
@@ -163,156 +178,147 @@ export function installHook(target: any): DevToolsHook | null {
-          : '' + maybeMessage;
-
-      // If the first argument is a string, check for substitutions.
-      if (typeof maybeMessage === 'string') {
-        if (args.length) {
-          const REGEXP = /(%?)(%([jds]))/g;
-
-          formatted = formatted.replace(REGEXP, (match, escaped, ptn, flag) => {
-            let arg = args.shift();
-            switch (flag) {
-              case 's':
-                arg += '';
-                break;
-              case 'd':
-              case 'i':
-                arg = parseInt(arg, 10).toString();
-                break;
-              case 'f':
-                arg = parseFloat(arg).toString();
-                break;
-            }
-            if (!escaped) {
-              return arg;
-            }
-            args.unshift(arg);
-            return match;
-          });
-        }
-      }
+    const args = inputArgs.slice();
 
-      // Arguments that remain after formatting.
-      if (args.length) {
-        for (let i = 0; i < args.length; i++) {
-          const arg = args[i];
+    // Symbols cannot be concatenated with Strings.
+    let formatted: string =
+      typeof maybeMessage === 'symbol'
+        ? maybeMessage.toString()
+        : '' + maybeMessage;
 
-          // Symbols cannot be concatenated with Strings.
-          formatted += ' ' + (typeof arg === 'symbol' ? arg.toString() : arg);
-        }
+    // If the first argument is a string, check for substitutions.
+    if (typeof maybeMessage === 'string') {
+      if (args.length) {
+        const REGEXP = /(%?)(%([jds]))/g;
+
+        formatted = formatted.replace(REGEXP, (match, escaped, ptn, flag) => {
+          let arg = args.shift();
+          switch (flag) {
+            case 's':
+              arg += '';
+              break;
+            case 'd':
+            case 'i':
+              arg = parseInt(arg, 10).toString();
+              break;
+            case 'f':
+              arg = parseFloat(arg).toString();
+              break;
+          }
+          if (!escaped) {
+            return arg;
+          }
+          args.unshift(arg);
+          return match;
+        });
       }
+    }
 
-      // Update escaped %% values.
-      formatted = formatted.replace(/%{2,2}/g, '%');
+    // Arguments that remain after formatting.
+    if (args.length) {
+      for (let i = 0; i < args.length; i++) {
+        const arg = args[i];
 
-      return '' + formatted;
+        // Symbols cannot be concatenated with Strings.
+        formatted += ' ' + (typeof arg === 'symbol' ? arg.toString() : arg);
+      }
     }
 
-    return '';
+    // Update escaped %% values.
+    formatted = formatted.replace(/%{2,2}/g, '%');
+
+    return '' + formatted;
   }
 
-  // NOTE: KEEP IN SYNC with src/backend/console.js:patch
-  function patchConsoleForInitialRenderInExtension(
+  let unpatchFn = null;
+
+  // NOTE: KEEP IN SYNC with src/backend/console.js:patchForStrictMode
+  function patchConsoleForInitialRenderInStrictMode(
```

Comment:
This function could use a comment block explaining _why_ it exists.

I _think_ the reason is specifically to handle the render (or renders) that happen while React and React DevTools are still connecting (when a renderer has _injected_ itself but hasn't yet been paired with a "renderer interface" on the DevTools side).

So this patching of the console only exists to do a very specific thing: hide or dim console logs during strict mode double render.